examples/nidcpower_source_dc_voltage/_helpers.py

```diff
@@ -231,3 +236,40 @@ def use_simulation_option(default: bool) -> Callable[[F], F]:
         is_flag=True,
         help="Use simulated instruments.",
     )
+
+
+def get_grpc_device_channel(
+    measurement_service: MeasurementService, driver_module: types.ModuleType
+) -> grpc.Channel:
+    """Returns driver specific grpc device channel."""
+    return measurement_service.get_channel(
+        provided_interface=getattr(driver_module, "GRPC_SERVICE_INTERFACE_NAME"),
+        service_class="ni.measurementlink.v1.grpcdeviceserver",
+    )
+
+
+def create_session_management_client(
+    measurement_service: MeasurementService,
+) -> nims.session_management.Client:
+    """Return created session management client."""
+    return nims.session_management.Client(
+        grpc_channel=measurement_service.get_channel(
+            provided_interface=nims.session_management.GRPC_SERVICE_INTERFACE_NAME,
+            service_class=nims.session_management.GRPC_SERVICE_CLASS,
+        )
+    )
+
+
+def get_session_and_channel_string(pin, sessions):
+    """Returns the session information for a given pin"""
+    pin_based_sessions = []
+    for session_info in sessions:
```

Comment:
It is very surprising for a parameter named `sessions` to contain `SessionInfo` objects and not `Session` objects.

If this doesn't return a `Session` object or an index that the caller can use to get the `Session` object, I don't think it solves Jack's problem.

Reply:
This API is intended to be used before the creation of the session and the api returns only the session information.
making this return the actual session might make this more driver specific, to make it a generic implementation we are only returning the session information.
The new implementation in #276 , defines a much simpler method.

Reply:
> making this return the actual session might make this more driver specific

No, you can write a generic function that takes a Sequence[T] and returns a T.